spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe "customer show page" do 
+  it "displays customer's name, items name and price, name of supermarket" do 
+    kroger = Supermarket.create!(name: 'Kroger', location: '123 4th St., Nashville, TN 56789')
+    leah = Customer.create!(name: 'Leah')
+    food = kroger.items.create!(name: 'Kibble', price: 20)
+    treat = kroger.items.create!(name: 'Bones', price: 10)
+    CustomerItem.create!(customer: leah, item: food)
+    CustomerItem.create!(customer: leah, item: treat)
+
+    visit "/customers/#{leah.id}"
+
+    within ".customer-info" do 
+      expect(page).to have_content(leah.name)
+    end
+
+    within ".item-info" do 
```

Comment:
Yeah, I like that you used the within block here, but we can be even more explicit and targeted by creating a unique css id for each item and it's own attributes. That's what I believe you were talking about in your summary and I recommend doing that so we can make sure the Kibble is being listed with the price of 20 and not the price of 10.